packages/flutter_tools/lib/src/commands/clean.dart

```diff
@@ -28,38 +32,71 @@ class CleanCommand extends FlutterCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
+    // Clean Xcode to remove intermediate DerivedData artifacts.
+    // Do this before removing ephemeral directory, which would delete the xcworkspace.
+    final FlutterProject flutterProject = FlutterProject.current();
+    if (xcode.isInstalledAndMeetsVersionCheck) {
+      await _cleanXcode(flutterProject.ios);
+      await _cleanXcode(flutterProject.macos);
+    }
+
     final Directory buildDir = fs.directory(getBuildDirectory());
-    _deleteFile(buildDir);
+    deleteFile(buildDir);
 
-    final FlutterProject flutterProject = FlutterProject.current();
-    _deleteFile(flutterProject.dartTool);
+    deleteFile(flutterProject.dartTool);
 
     final Directory androidEphemeralDirectory = flutterProject.android.ephemeralDirectory;
-    _deleteFile(androidEphemeralDirectory);
+    deleteFile(androidEphemeralDirectory);
 
     final Directory iosEphemeralDirectory = flutterProject.ios.ephemeralDirectory;
-    _deleteFile(iosEphemeralDirectory);
+    deleteFile(iosEphemeralDirectory);
+
+    final Directory macosEphemeralDirectory = flutterProject.macos.ephemeralDirectory;
+    deleteFile(macosEphemeralDirectory);
 
     return const FlutterCommandResult(ExitStatus.success);
   }
 
-  void _deleteFile(FileSystemEntity file) {
-    final String path = file.path;
-    printStatus("Deleting '$path${fs.path.separator}'.");
-    if (file.existsSync()) {
-      try {
-        file.deleteSync(recursive: true);
-      } on FileSystemException catch (error) {
-        if (platform.isWindows) {
-          printError(
-            'Failed to remove $path. '
+  Future<void> _cleanXcode(XcodeBasedProject xcodeProject) async {
+    if (!xcodeProject.existsSync()) {
+      return;
+    }
+    Status xcodeStatus;
+    try {
+      final Directory xcodeWorkspace = xcodeProject.xcodeWorkspace;
+      final XcodeProjectInfo projectInfo = await xcodeProjectInterpreter.getInfo(xcodeWorkspace.parent.path);
+      for (String scheme in projectInfo.schemes) {
+        xcodeStatus = logger.startProgress('Cleaning Xcode workspace for scheme $scheme...', timeout: timeoutConfiguration.slowOperation);
```

Comment:
I'm seeing no newlines after the spinners, which made the output really weird:
```
$ flutter clean
Cleaning Xcode workspace for scheme Build Flutter Bundle...            ⣽Deleting build...                                                  105ms
Deleting ephemeral...                                                4ms
       ⣻$
```

Reply:
I just pushed a fix for the progress part, but I suspect the workspace clean is failing.  Can you try again with `flutter clean -v`?

Reply:
This is the workspace section of the `-v` output (using the FDE example app; full paths elided):
```
[ +169 ms] Cleaning Xcode workspace...
[   +5 ms] executing: [.../example/macos/] /usr/bin/xcodebuild -list
[+1031 ms] Information about project "Runner":
                        Targets:
                            Runner
                            Build Flutter Bundle

                        Build Configurations:
                            Debug
                            Release

                        If no build configuration is specified and -scheme is not passed then
                        "Release" is used.

                        Schemes:
                            Build Flutter Bundle
                            Runner
[   +2 ms] executing: [.../example/] /usr/bin/xcodebuild -workspace
.../example/macos/Runner.xcworkspace -scheme Build Flutter Bundle -quiet clean
[+1082 ms] Exit code 0 from: /usr/bin/xcodebuild -workspace
.../example/macos/Runner.xcworkspace -scheme Build Flutter Bundle -quiet clean
[   +1 ms] executing: [.../example/] /usr/bin/xcodebuild -workspace
.../example/macos/Runner.xcworkspace -scheme Runner -quiet clean
[+1088 ms] Exit code 0 from: /usr/bin/xcodebuild -workspace
.../example/macos/Runner.xcworkspace -scheme Runner -quiet clean
```

Reply:
That looks correct. Do you still see the lack of newlines without `-v`?

Reply:
No, formatting without -v looked good in that test.